www/js/app.js

```diff
@@ -55,10 +52,44 @@ var app = angular.module('starter',
                 url:'/settings',
                 templateUrl: 'templates/settings.html',
                 controller: 'Settings'
+            })
+            .state('profile', {
+                url:'/profile',
+                templateUrl: 'templates/profile.html',
+                controller: 'Profile'
             });
         $urlRouterProvider.otherwise('/');
     }) 
     
+    .directive('profileBarWidget', function($ionicGesture) {
+        return {
+            restrict: 'AE',
+            // declare the directive scope as private (and empty)
+            scope: {},
+            // add behaviour to our buttons and use a variable value
+            template:   '<div class="thumbnail">'+
+                            '<img src="img/ionic.png" style="width:60px;height:60px;"'+
+                                'alt="Not By Design" />'+
+                        '</div>'+
+                        '<div class="profile-name">'+       
+                            '<h3> {{ profile.name }} </h3>'+
+                        '</div>'+
+                        '<div class="edit-button">'+
+                            '<button class="button button-positive button-small"'+
+                               'onclick="window.location=\'#/profile\'">Edit</button>'+
+                        '</div>',
+            // we just declare what we need in the above template
+            controller: function(userInfoService, $scope) {
+                userInfoService.getProfileInfo().then(function(userInfo) {
```

Comment:
I've tried moving the controller to a seperate file, but i get errors running the app after that 
(Error: [ng:areq] Argument 'ProfileBarController' is not a function, got undefined) If you can let me know how to do this. Same with loading the template from a seperate html file.